crates/libcgroups/src/v2/devices/bpf.rs

```diff
@@ -11,13 +11,33 @@ pub mod prog {
     use std::os::unix::io::RawFd;
     use std::ptr;
 
-    pub fn load(license: &str, insns: &[u8]) -> Result<RawFd> {
-        let insns_cnt = insns.len() / std::mem::size_of::<libbpf_sys::bpf_insn>();
-        let insns = insns as *const _ as *const libbpf_sys::bpf_insn;
+    use libbpf_sys::{bpf_insn, BPF_CGROUP_DEVICE, BPF_F_ALLOW_MULTI, BPF_PROG_TYPE_CGROUP_DEVICE};
+    use libc::{rlimit, ENOSPC, RLIMIT_MEMLOCK};
+
+    // mocks
+    // TODO: consider use of #[mockall_double]
+    #[cfg(test)]
+    use crate::v2::devices::mocks::mock_libc::setrlimit;
+    #[cfg(not(test))]
+    use libc::setrlimit;
+
+    // TODO: consider use of #[mockall_double]
+    #[cfg(test)]
+    use crate::v2::devices::mocks::mock_libbpf_sys::{
+        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+    };
+    #[cfg(not(test))]
+    use libbpf_sys::{
+        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+    };
 
+    pub fn load(license: &str, insns: &[u8]) -> Result<RawFd> {
+        let insns_cnt = insns.len() / std::mem::size_of::<bpf_insn>();
+        let insns = insns as *const _ as *const bpf_insn;
+        #[allow(unused_unsafe)]
```

Comment:
I don't like this one, but the mocked stuff isn't actually unsafe.

We could probably make it look `unsafe` though...